### DISCUSSION TITLE
Possible to fill in the area between two Y values? A filled trend-line?

### DISCUSSION THREAD

**Discussion by Tyler-Lopez** [patrykandpatrick/vico#855] [Questions] [answered]
### Question

Is this type of functionality possible, where every X value has two Y values and we fill in the area created from this? I was thinking it would be possible with PointConnector but we of course only get two Y coordinates there, not the four we would need.

![Screenshot 2024-08-19 at 1 12 08 PM](https://github.com/user-attachments/assets/573215b5-1633-4d7e-8d06-834fdee51628)

### UI framework(s)

Jetpack Compose

**Comment by Tyler-Lopez** [comment]
I can think of quite a complicated, ugly way to do this, by creating two line series and overriding the area fill with a custom implementation in both cases. I'd love to know of other ideas.

**Reply by Gowsky** [reply, marked as answer]
Hello, @Tyler-Lopez. As mentioned in #893, you can achieve this through subclassing, and we plan on adding such a chart type in the future. A custom `AreaFill` implementation would probably work too, but subclassing `LineCartesianLayer` will give you better access to the required data and more flexibility in general.

**Reply by Tyler-Lopez** [reply]
Hi there, thanks for linking the other discussion, I'll take a look!

I haven't touched this in some time, but if you are interested, this is how I achieved the trend line in my explorations of this library.

It was two `LineCartesianLayer` (not that it had to be) where the first was not drawn, and only served to create the bottom boundary. The second layer then would use that bottom boundary.

https://gist.github.com/Tyler-Lopez/f96adeb92a663928cc77d6fc202a7b39
